catalog/ui/catalog-ui-search/src/main/webapp/js/workers/worker.less.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import {BaseWorker} from './worker.base';
+
+(function prepLessForWebWorker() {
+    global.window = self;
+    global.window.less = {
+        onReady: false
+    };
+    global.window.document = {
+        getElementsByTagName: function (tagName) {
+            if (tagName === 'script') {
+                return [{
+                    dataset: {}
+                }];
+            } else if (tagName === 'style') {
+                return [];
+            } else if (tagName === 'link') {
+                return [];
+            }
+        }
+    };
+})();
+
+const Less = require('less');
+const variableRegex = '/@(.*:[^;]*)/g';
```

Comment:
Eventually it might be nice for these to be refined since they seem a little broad as is.